### ISSUE TITLE
Need tester

### ISSUE THREAD

**Issue by riderkick** [riderkick/FMD#77]
Before releasing the new version I need you to test this changes:
- [x] Now we are using sqlite3 database for mangalist. I need you to test opening mangalist, old data will be converted automatically when opening the data, it may take a moment for large data. Please test to build mangalist from scratch by deleting the existing data (*.dat, you can backup it first). Recently I encounter "Out of Memory" when building mangalist from scratch, it's due to fmd using to much RAM when building mangalist, because the data is processed in RAM only. This is the reason I convert the data process to database.
- [x] Opening/searching mangalist now using separated thread, It's only using 1 thread, so other query will be queued to this thread, I need you to test with fast type.
- [x] Test to use fmd with unicode character in fmd path.
- [x] Test to shutdown the system with fmd, previously there is an issue that freeze fmd in the middle of shutdown counter.
- [x] Test the new site mh$160, since it's too slow from my place there is no mangalist, try to build from scratch, it also will help to test new mangalist mechanism.

Please report if you found any problem. Complete with log, additionally there will be "log" folder. If this log don't created you can trigger it by opening fmd with this param `fmd.exe -LOGACTIVE 4`

Download [here](http://www.mediafire.com/download/c9tcpkb9sqcat8t/fmd_0.9.x.x.7z)

**Comment by abhi98** [comment]
the new site mh$160 is working good and shutting down the system with FMD works with no issue. using FMD with Unicode character working good. I don't know what you mean in first two test sorry.

**Comment by riderkick** [comment]
Thanks for your report.
Previously we use raw data for mangalist, If you look at the folder "data", the data which usually *.dat will be changed to *.db.
Just do you usual stuff, changing mangalist and do some search with fast typing. If you have fast internet connection you may can build mangalist from scratch by deleting the existing file (*.dat/*.db, you can backup it first), do it with several website.
Just report if you have any problem.

**Comment by abhi98** [comment]
manga-list of R e a d m a n g a . t o d a y doesn't get updated, it gets stuck at "removing duplicate from current data"  only in this version and I did what you told me, I tried with 15 sites and searched with fast typing and didn't have any  problem.

**Comment by abhi98** [comment]
I have encountered a problem when downloading from mh$160. the task would failed because of downloading only some pages of the chapter, not all of pages. I had to resume the task for atleast 15 times to get all the chapter downloaded. please try to fix it. Thanks :)

the like to the manhua:- http://www.mh160.com/kanmanhua/24063/

**Comment by riderkick** [comment]
Did you build the manga list from scratch or just update it?
If you just update it, there is slow locate function when "removing duplicate from current data". In large data it may appear "stuck", but actually it just slow. I replace it with much more faster approach https://github.com/riderkick/FMD/commit/59be9b7527349acd4b4809596068fa01b6c6e905.

**Comment by abhi98** [comment]
I just updated the manga list.

**Comment by riderkick** [comment]
mh$160 generate the real image url with javascript in the DOM, which is hard to follow, maybe there is different rule for each chapter/series.

**Comment by abhi98** [comment]
I don't think there is different rule for each chapter and/or series.

**Comment by riderkick** [comment]
The file name `_20110501` suggest there is another version. 
I'll try to rewrite this script in pascal.

**Comment by riderkick** [comment]
I've been update the link, please try the new build.

**Comment by abhi98** [comment]
oh so that's what you meant. i will the new build

**Comment by abhi98** [comment]
The task failed the first time. then I resume the task. there were 4 chapters that weren't downloaded. I resume the task to see if they would be downloaded or not. the task gets stuck at 0bps when downloading those 4 chapters.

**Comment by riderkick** [comment]
You should mention which title and chapter.

**Comment by abhi98** [comment]
the link: http://www.mh160.com/kanmanhua/11105/
its was four chapters that didn't get downloaded and one of them was chapter 87

**Comment by riderkick** [comment]
![sshot-1](https://cloud.githubusercontent.com/assets/4972676/8883269/1b301628-3280-11e5-85cb-67ddaad7257c.jpg)

**Comment by riderkick** [comment]
Did you even try to open those chapter in browser?
Because on browser the image indeed can't be loaded. Whether the link is dead or I even can't reach the server at all.
![sshot-2](https://cloud.githubusercontent.com/assets/4972676/8883418/5f1ee73c-3281-11e5-80e6-16c053c454f4.jpg)

This is the url of images:
```
http://61.164.109.41:82/comic/D/%E6%96%97%E7%A0%B4%E8%8B%8D%E7%A9%B9/%E7%AC%AC87%E8%AF%9D/1.jpg
http://61.164.109.41:82/comic/D/%E6%96%97%E7%A0%B4%E8%8B%8D%E7%A9%B9/%E7%AC%AC87%E8%AF%9D/2.jpg
http://61.164.109.41:82/comic/D/%E6%96%97%E7%A0%B4%E8%8B%8D%E7%A9%B9/%E7%AC%AC87%E8%AF%9D/2.jpg
http://61.164.109.41:82/comic/D/%E6%96%97%E7%A0%B4%E8%8B%8D%E7%A9%B9/%E7%AC%AC87%E8%AF%9D/3.jpg
```

**Comment by abhi98** [comment]
Yes I did, before I posted about not being to able to download the four chapter, and I thought that it was only my browser that didn't load the images. Do you know the way of knowing what four chapter didn't get downloaded or do I have to check every folder ?

**Comment by riderkick** [comment]
After the task finished and there is failed chapter, it will replace the main chapterlist. So if you stop the cursor on top of the task in the hint it will show those failed chapters. Or you can try to open works.ini, there is failedchapterlinks, or something like that if the task not finished yet.

**Comment by abhi98** [comment]
images load up in browser but FMD can't download the chapter. i am trying to download both parts of chapter 41. Please help. Thanks 

the link: http://www.mh160.com/kanmanhua/22384/

**Comment by riderkick** [comment]
I update the link, please try

**Comment by blue-lightning** [comment]
Done the first test, report now:
I made a fresh list off of b a t o t o, info getting speed averaged to ~1000 entries/h. There's current 14057, so 14 hours are needed. I tried for 8 continuous hours with no problem before my network was cut off by a power glitch.
Then FMD started running through the remaining entries at breakneck speed until it's "done" instead of displaying a warning or just destroy the thread like I expected. Furthermore, the list is broken and no entry showed up in the mangalist column, even though ~8000 first entries are gotten normally.

**Comment by riderkick** [comment]
How it's broken if it can show ~8000 first entries? Does the remaining entries doesn't have title?

**Comment by blue-lightning** [comment]
Sorry, what I meant is ~8000 entries should be gotten normally (the rest is probably broken, I stopped connecting to the net after all) and stores in the database, but they are not displayed in the column. Currently the mangalist column is still in a blank state.

**Comment by riderkick** [comment]
is there batoto.db in data folder? And what about the content, does it have any record?

**Comment by blue-lightning** [comment]
No. I thought everything is now stored in fmd.dbg ?
Does that mean those ~8000 entries aren't stored in my system but were discarded instead when the network failed ?

**Comment by riderkick** [comment]
fmd.dbg is used for debugging, its contain debug information.
It will stored to disk every 32 record added to database. So if your power cut off and your computer suddenly died, you should only lose 32 record at max.
Try to build again, and abort the update after few title loaded. It should save at least that much.

**Comment by blue-lightning** [comment]
Abort update, shutdown FMD, cut off internet. Tried 3 kinds of interruption. None results in a partial working mangalist column. While it still is blank, it doesn't ask me to download b a t o t o manga list from server anymore.

**Comment by riderkick** [comment]
It will only ask if you change selected website.
Ba to to is very slow. Try to build another website data.
I just visit ba to to, it seems now they have sorted list, I'll try to rewrite the ba to to script from scratch tomorrow.

**Comment by blue-lightning** [comment]
Just update to 0.9.19.2, everything works now. FMD saves whatever it got correctly before aborting. Thanks for being patient with me.

**Comment by riderkick** [comment]
Now ba to to using ordered list added time, it will not saved if you abort the update, because we may lose some title if we force it to save the data. Ordered list is faster for update, because it will stop at first found link that exist on database, and just load what needed.
